composable_kernel/include/tensor_operation/gridwise_2d_reduction_multiblock_two_call.hpp

```diff
@@ -0,0 +1,472 @@
+#include "reduction_operator.hpp"
+#include "reduction_functions_accumulate.hpp"
+#include "reduction_functions_blockwise.hpp"
+
+#include "threadwise_tensor_slice_transfer.hpp"
+
+namespace ck {
+
+template <typename GridwiseReduction,
+          bool NeedIndices,
+          typename InDataType,
+          typename AccDataType,
+          typename In2dDescType,
+          typename Workspace2dDescType,
+          typename InElementwiseOperation,
+          typename AccElementwiseOperation>
+__global__ void kernel_reduce_multiblock_two_call(const In2dDescType in2dDesc,
+                                                  const Workspace2dDescType ws2dDesc,
+                                                  const InElementwiseOperation in_elementwise_op,
+                                                  const AccElementwiseOperation acc_elementwise_op,
+                                                  int BlkGroupSize,
+                                                  int kBlockTileIterations,
+                                                  const InDataType* const __restrict__ p_src_global,
+                                                  AccDataType* const __restrict__ ws_values_global,
+                                                  int* const __restrict__ ws_indices_global)
+
+{
+    if constexpr(!NeedIndices)
+        GridwiseReduction::Run(in2dDesc,
+                               ws2dDesc,
+                               in_elementwise_op,
+                               acc_elementwise_op,
+                               BlkGroupSize,
+                               kBlockTileIterations,
+                               p_src_global,
+                               ws_values_global,
+                               ws_indices_global);
+    else
+        GridwiseReduction::RunWithIndices(in2dDesc,
+                                          ws2dDesc,
+                                          in_elementwise_op,
+                                          acc_elementwise_op,
+                                          BlkGroupSize,
+                                          kBlockTileIterations,
+                                          p_src_global,
+                                          ws_values_global,
+                                          ws_indices_global);
+};
+
+template <typename InDataType,
+          typename dstDataType,
+          typename AccDataType,
+          typename In2dDescType,
+          typename Workspace2dDescType,
+          typename ReduceOperation,
+          typename InElementwiseOperation,
+          typename AccElementwiseOperation,
+          bool PropagateNan,
+          index_t BlockSize,
+          index_t MThreadClusterSize,
+          index_t KThreadClusterSize,
+          index_t MThreadSliceSize,
+          index_t KThreadSliceSize,
+          index_t InSrcVectorDim,
+          index_t InSrcVectorSize,
+          index_t OutDstVectorSize>
+struct GridwiseReduction_xy_to_x_multiblock_two_call
```

Comment:
please be consistent about the naming of dimensions
M: non-reduce dimension
K: reduction dimension

Reply:
Agree